airflow/providers/amazon/aws/example_dags/example_redshift.py

```diff
@@ -0,0 +1,67 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This is an example dag for using `RedshiftSQLOperator` to authenticate with Amazon Redshift
+then execute a simple select statement
+"""
+# [START redshift_operator_howto_guide]
+from airflow import DAG
+from airflow.providers.amazon.aws.operators.redshift import RedshiftSQLOperator
+from airflow.utils.dates import days_ago
+
+with DAG(dag_id="example_redshift", start_date=days_ago(1), schedule_interval=None, tags=['example']) as dag:
+    # [START howto_operator_redshift_create_table]
+    setup__task_create_table = RedshiftSQLOperator(
+        task_id='setup__create_table',
+        sql="""
+            CREATE TABLE IF NOT EXISTS fruit (
+            fruit_id INTEGER,
+            name VARCHAR NOT NULL,
+            color VARCHAR NOT NULL
+            );
+        """,
+    )
+    # [END howto_operator_redshift_create_table]
+    # [START howto_operator_redshift_populate_table]
+    task_insert_data = RedshiftSQLOperator(
+        task_id='task_insert_data',
+        sql=[
+            "INSERT INTO fruit VALUES ( 1, 'Banana', 'Yellow');",
+            "INSERT INTO fruit VALUES ( 2, 'Apple', 'Red');",
+            "INSERT INTO fruit VALUES ( 3, 'Lemon', 'Yellow');",
+            "INSERT INTO fruit VALUES ( 4, 'Grape', 'Purple');",
+            "INSERT INTO fruit VALUES ( 5, 'Pear', 'Green');",
+            "INSERT INTO fruit VALUES ( 6, 'Strawberry', 'Red');",
+        ],
+    )
+    # [END howto_operator_redshift_populate_table]
+    # [START howto_operator_redshift_get_all_rows]
+    task_get_all_table_data = RedshiftSQLOperator(
+        task_id='task_get_all_table_data', sql="CREATE TABLE more_fruit AS SELECT * FROM fruit;"
+    )
+    # [END howto_operator_redshift_get_all_rows]
+    # [START howto_operator_redshift_get_with_filter]
+    task_get_with_filter = RedshiftSQLOperator(
+        task_id='task_get_with_filter',
+        sql="CREATE TABLE filtered_fruit AS SELECT * FROM fruit WHERE color = '{{ params.color }}';",
+        params={'color': 'Red'},
```

Comment:
`params` is a BaseOperator thing -- I think you want `parameters`, and this:

```suggestion
        sql="CREATE TABLE filtered_fruit AS SELECT * FROM fruit WHERE color = :color;",
        parameters={'color': 'Red'},
```

Reply:
Oh that only works if the `paramstyle` is set on the connection. The default is:

```suggestion
        sql="CREATE TABLE filtered_fruit AS SELECT * FROM fruit WHERE color = %s;",
        parameterss=['Red'],
```

Reply:
@ash a question on this.

Are you saying that `params` is not meant for templating?  Or are you saying that in sql, we should prefer `parameters` _when possible_?

E.g. I have used `params` e.g. like this:

```sql
UNLOAD ('
SELECT *
FROM my_table a
')
TO 's3://{{ params.bucket }}/{{ params.prefix }}/'
```

I am pretty sure you can't use parameters for this kind of thing.  I had one other case where it was a massive union and I pulled the logic into a sql template generation script.

I reckon what you are saying is that a better practice to use `parameters` whenever it's supported, and in this case it's supported so that's what the example should show.  And in other cases no issue with params in templates.   

Side note, it's weird, i looked at BaseOperator and the docstring does not document `params` 🤔

Reply:
fixed to use `parameters` in 66ba154c6e5fdc87e36eb97054cf0ac22fd295ca